Assets/UXF/Scripts/Trackers/ScreenRayTracker.cs

```diff
@@ -0,0 +1,150 @@
+    public bool useLayerMask = false;
+    [Tooltip("Provide the names of the layers for the mask.")]
+    public string[] layerMaskNames;
+
+    // Private vars
+    private List<string> objectDetected = new List<string>();
+    private UXFDataRow currentRow;
+    private bool recording = false;
+    private int layerMask;
+    private string noObjectString = "NA";
+    private int numRays;
+    private List<float> x  = new List<float>();
+    private List<float> y  = new List<float>();
+
+    // Start calc
+    void Start(){
+        // Create layer mask 
+        if(useLayerMask){
+            layerMask = LayerMask.GetMask(layerMaskNames);
+        } else {
+            layerMask = ~0; // Set to everything as no mask is wanted. 
+        }
+
+    	// Start the recoding
+        StartCoroutine(RecordRoutine());
+    }
+
+    /// <summary>
+    /// Gets coordinates for the rays in screen space from .json file and prints screen resolution
+    /// </summary>
+    public void GetRayCoordinates(){
+        // Coordinates
+        x = session.settings.GetFloatList("ray_x");
+        y = session.settings.GetFloatList("ray_y");
+
+        // Screen resolution
+        Debug.Log(Screen.currentResolution);
+    }
+
+    /// <summary>
+    /// Starts the recording. This method needs to be added to [UXF_Rig] events called On Trial Begin
+    /// </summary>
+    public void StartRecording(){
+    	recording = true;
+    }
+
+    /// <summary>
+    /// Stops the recording. This method needs to be added to [UXF_Rig] events called On Trial End
+    /// </summary>
+    public void StopRecording(){
+    	recording = false;
+    }
+
+    IEnumerator RecordRoutine(){
+        while (true){
+            if (recording){
+                objectDetected = ray2detectObjects(x, y, cam);
+                for(int i = 0; i < numRays; i++){
+                    // When no object was detected save only if saveNoObject is true
+                    if(objectDetected[i] != noObjectString){
+                        var values = new UXFDataRow();
+                        values.Add(("rayIndex", i));
+                        values.Add(("x", x[i]));
+                        values.Add(("y", y[i]));
+                        values.Add(("objectDetected", objectDetected[i]));
+                        currentRow = values;
+                        RecordRow(); // record for each ray
+                        currentRow = null;
+
+                    } 
+                }
+            }
+            yield return null; // wait until next frame
+        }
+    }
+
+    /// <summary>
+    /// Set headers and measurment descriptor
+    /// </summary>
+    public override string MeasurementDescriptor => "ObjectsOnScreenTracker";
+    public override IEnumerable<string> CustomHeader => new string[] { "rayIndex", "x", "y", "objectDetected"};
+
+    // Get values
+    protected override UXFDataRow GetCurrentValues(){
+        return currentRow;
+    }
```

Comment:
this means we will end up recording the values twice?

Reply:
Mhmm, strange in the result files each row is unique as far as I can see.

Reply:
what if you use `UpdateMode` that is not manual?